File: HackerRank Solutions/30 Days of Code/Day 11 2D Arrays/Day 11 2D Arrays.java
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.regex.*;

public class Solution {



    private static final Scanner scanner = new Scanner(System.in);

    public static void main(String[] args) {
        int[][] arr = new int[6][6];

        for (int i=0; i<6; i++){
            String[] arrRowItems = scanner.nextLine().split(" ");
            scanner.skip("(\r\n|[\n\r\u2028\u2029\u0085])?");

            for (int j=0; j<6; j++){
                int arrItem = Integer.parseInt(arrRowItems[j]);
                arr[i][j] = arrItem;
            }
        }
        int sum=0, maxSum=-99999;
        for (int i=1; i<5; i++){
            for (int j=1; j<5; j++){
                sum = arr[i][j] + arr[i-1][j] + arr[i-1][j-1] + arr[i-1][j+1] + arr[i+1][j] + arr[i+1][j-1] + arr[i+1][j+1];
                if (maxSum < sum){
                    maxSum = sum;
                }
            }
        }
        System.out.println(maxSum);
        scanner.close();
    }
}

File: HackerRank Solutions/30 Days of Code/Day 11 2D Arrays/Day 11 2D Arrays.py
#!/bin/python3

import math
import os
import random
import re
import sys



if __name__ == '__main__':
    arr = []

    for _ in range(6):
        arr.append(list(map(int, input().rstrip().split())))
    sum = 0
    maxSum = -9 * 7
    for i in range(len(arr) - 2):
        for j in range(len(arr[i]) - 2):
            sum = arr[i][j] + arr[i][j+1] + arr[i][j+2] + arr[i+1][j+1] + arr[i+2][j] + arr[i+2][j+1] + arr[i+2][j+2]
            maxSum = max(sum, maxSum)
    print(maxSum)
